src/staking.rs

```diff
@@ -113,10 +113,19 @@ pub enum StakingSudo {
     /// Causes the unbonding queue to be processed.
     /// This needs to be triggered manually, since there is no good place to do this right now.
     /// In cosmos-sdk, this is done in `EndBlock`, but we don't have that here.
+    #[deprecated(note = "This is not needed anymore. Just call `update_block`")]
     ProcessQueue {},
 }
 
-pub trait Staking: Module<ExecT = StakingMsg, QueryT = StakingQuery, SudoT = StakingSudo> {}
+pub trait Staking: Module<ExecT = StakingMsg, QueryT = StakingQuery, SudoT = StakingSudo> {
+    fn process_queue<ExecC, QueryC: CustomQuery>(
```

Comment:
Since this function is now part of the trait people have to implement in order to supply their own staking module: Could you add a doc comment so that potential implementors know what to do here / when it's called?

Reply:
Sure. I was thinking, we can also provide a no-op default impl. So that implementing a dummy version of the staking module would be simpler. WDYT?

Reply:
I am not so sure about the default impl. It would make this a non-breaking change, but I feel like a default impl should only be provided if there is a reasonable default behaviour. Because of this:
> The disadvantage is of course that a more or less 'real' stack module impl would need this.

I would say it doesn't make sense as a default behaviour here. If people want a dummy version, they can just implement the function with an empty body themselves, making it explicit.